atomicfu/src/jvmMain/kotlin/kotlinx/atomicfu/parking/JvmParkingDelegator.kt

```diff
@@ -0,0 +1,43 @@
+package kotlinx.atomicfu.parking
+import java.util.concurrent.atomic.AtomicLong
+import java.util.concurrent.locks.LockSupport
+import kotlin.time.DurationUnit
+import kotlin.time.TimeSource.Monotonic
+
+// Only for testing purposes
+internal class JvmParkingDelegator: ParkingDelegator {
+    private var thread: Thread? = null
+    private val atomicLong: AtomicLong = AtomicLong(0L)
```

Comment:
Let's rename it to state, document that these states are "PARKED" and "UNPARKED", and use either `AtomicBoolean` provided AtomicFU or `AtomicIntegerFieldUpdater` if this is easier due to the compilation complexity.

Reply:
Moved all state to ParkingData similar to native